src/client.js

```diff
@@ -34,38 +34,40 @@ function initSocket() {
 
 global.socket = initSocket();
 
-const component = (
-  <Router render={(props) =>
-        <ReduxAsyncConnect {...props} helpers={{client}} filter={item => !item.deferred} />
-      } history={history}>
-    {getRoutes(store)}
-  </Router>
-);
+const routes = getRoutes(store);
 
-ReactDOM.render(
-  <Provider store={store} key="provider">
-    {component}
-  </Provider>,
-  dest
-);
-
-if (process.env.NODE_ENV !== 'production') {
-  window.React = React; // enable debugger
-
-  if (!dest || !dest.firstChild || !dest.firstChild.attributes || !dest.firstChild.attributes['data-react-checksum']) {
-    console.error('Server-side React render was discarded. Make sure that your initial render does not contain any client-side code.');
-  }
-}
+match({ history, routes }, (error, redirectLocation, renderProps) => {
+  const component = (
```

Comment:
if server response with status 500, app goes cyclic redirect because render props would be undefined, therefore hydrateOnClient don't work correctly in error case